client/src/components/Cards/cards.tsx

```diff
@@ -22,9 +23,21 @@ export const Cards = () => {
 		dialogRef.current?.close();
 		close();
 	};
+	  const router = useRouter();
+		const { pathname, query } = router;
+
+		// Construct the new query parameters
+		const newQuery = { ...query, showModal: "y" };
+
+		// Create the href with the current pathname and updated query parameters
+		const href = {
+			pathname,
+			query: newQuery,
+		};
+
 	return (
 		<div className={styles.cardContainer}>
-			<Link className={styles.cardBody} href={"?showModal=y"}>
+			<Link className={styles.cardBody} href={href}>
```

Comment:
omg! Would this change fix the issue when we open the modal and close it reset the URL?

Reply:
This update doesn't directly handle closing the modal yet.

Reply:
That's fine! I was just too excited 😅